### ISSUE TITLE
Plugins incompatible with lazy loading

### ISSUE THREAD

**Issue by pjeby** [ohm-en/obsidian-plugin-manager#1]
Just a heads-up, since this issue has also happened with a similar plugin (see https://github.com/Mocca101/obsidian-plugin-groups/issues/12): at least the Pane Relief and Media Extended plugins don't work with lazy loading, and the beta of Workspaces Plus is unlikely to function correctly either.  Basically any plugin that relies on replacing built-in views or that stores state information in the workspace.

**Comment by ohm-en** [comment]
Thanks for being this to my attention. I figured there would be problems like this. You can't expect plugins to work when they're disabled after all, but you do make a good point in adding a warning. For a few severe cases, I can add them to the `id` blacklist, but for most I'd like to add a warning or event.

@pjeby, seeing as you're a more seasoned user of the Obsidian API, are you aware of a way to reload a workspace or detect when a plugin uses one?

P.S.
I had no idea another plugin this similar to mine released recently. Huh...

**Comment by pjeby** [comment]
Reloading the workspace won't help in Pane Relief's case, because the stored data will already be lost and overwritten.  (Same I expect for the Workspaces plus beta.)  Until Obsidian has its own API for plugin-specific workspace-specific storage, or until plugins like this and the other one have a way to keep the data around for them, they just can't be late-loaded.

(If you care about the technical details, some plugins store info with the workspace, which Obsidian doesn't really support on its own.  They work fine as long as they're loaded before the workspace is, because they can patch Obsidian's workspace loading code to capture the data.  But if that doesn't happen, the data isn't loaded, and Obsidian will overwrite the saved workspace with a fresh copy after loading, thus erasing it from disk.  This destroys settings for plugins that have per-workspace settings, and also erases the navigation history for Pane Relief.)

**Comment by ohm-en** [comment]
I appreciate the advice Pjeby. I'll be considering a few changes with all that in mind, but this isn't priority at the moment.

**Comment by kngautham** [comment]
Everytime I set lazy loading for Periodic Notes Plugin, Once i close and reopen the app, the periodic notes loads at startup for some reason.

**Comment by legolasdimir** [comment]
If a plugin doesnt show up in the toggle list does this mean it does not support lazy loading?

**Comment by ohm-en** [comment]
> If a plugin doesnt show up in the toggle list does this mean it does not support lazy loading?

No, plugins that aren't supported should show up as grayed out unless they're a core plugin (which aren't currently support).

If you're having a specific issue, would you mind opening a new issue with more details?
